### PR TITLE
Bd 090

### PR DESCRIPTION
**Pull request by BDonnot**
PR for issues 83, 84, 86, 87, 88 and 89 + new redispatching strategy

### CODE REVIEW

File: grid2op/Agent/TopologyGreedy.py [hunk 1]
```diff
@@ -28,7 +28,8 @@ def __init__(self, action_space):
     def _get_tested_action(self, observation):
         if self.li_actions is None:
             res = [self.action_space({})]  # add the do nothing
-            res += self.action_space.get_all_unitary_topologies_change(self.action_space)
+            # better change are still "bugged" for now
```
Comment (Tezirg) [inline grid2op/Agent/TopologyGreedy.py]: What do you mean ?

File: grid2op/Environment/Environment.py [hunk 2]
```diff
@@ -550,6 +552,7 @@ def reset(self):
                                          names_chronics_to_backend=self.names_chronics_to_backend)
         self.current_obs = None
         self.env_modification = None
+        super().reset()
```
Comment (Tezirg) [inline grid2op/Environment/Environment.py]: Maybe that should be called first, what if Parent class resets the chronics_handler ?

File: grid2op/Parameters.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ class Parameters:
         thermal
         limit. Default is ``False``
 
-    NB_TIMESTEP_POWERFLOW_ALLOWED: ``int``
+    DNB_TIMESTEP_OVERFLOW_ALLOWED: ``int``
```
Comment (Tezirg) [inline grid2op/Parameters.py]: Drum n Bass ? Or typo ?

File: grid2op/Agent/TopologyGreedy.py (hunk 1 above)
Comment (BDonnot) [reply grid2op/Agent/TopologyGreedy.py]: I am not 100% sure the "change" are all computed. Maybe we are still missing some, so better, in my opinion to have agent that "set" the stuff for the Greedy agent.

File: grid2op/Environment/Environment.py (hunk 2 above)
Comment (BDonnot) [reply grid2op/Environment/Environment.py]: yep you are right. I'll fix that :-)

File: grid2op/Parameters.py (hunk 3 above)
Comment (BDonnot) [reply grid2op/Parameters.py]: hahah it was a typo and the doc is not "unit tested" so i didn't catch that. I fixed it.

**Pull request merged**